### PR TITLE
MULE-13671: fix tests

### PR DESCRIPTION
**Pull request by marianogonzalez**

### CODE REVIEW

File: src/test/java/org/mule/test/extension/file/common/api/FileTestHarness.java [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+ * SFTP, etc) or the authentication method
+ *
+ * @since 1.0
+ */
+public interface FileTestHarness extends TestRule {
+
+  String HELLO_WORLD = "Hello World!";
+  String HELLO_FILE_NAME = "hello.json";
+  String BINARY_FILE_NAME = "binary.bin";
+  String HELLO_PATH = Paths.get("files/" + HELLO_FILE_NAME).toString();
```
Comment (pablokraan) [inline src/test/java/org/mule/test/extension/file/common/api/FileTestHarness.java]: Don't use / in the path element. Use Paths.get("files", HELLO_FILE_NAME)

File: src/test/java/org/mule/test/extension/file/common/api/FileTestHarness.java [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+  String DEFAULT_FTP_HOST = "localhost";
+  String FTP_SERVER_BASE_DIR = Paths.get("target/ftpserver").toString();
```
Comment (pablokraan) [inline src/test/java/org/mule/test/extension/file/common/api/FileTestHarness.java]: Same

File: src/test/java/org/mule/test/extension/file/common/api/FileTestHarness.java [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+  String WORKING_DIR_SYSTEM_PROPERTY = "workingDir";
+  String WORKING_DIR = "base";
+
+  /**
+   * Creates a test hello world file
+   */
+  void createHelloWorldFile() throws Exception;
```
Comment (pablokraan) [inline src/test/java/org/mule/test/extension/file/common/api/FileTestHarness.java]: Add missing @throws in all the javadocs

File: src/test/java/org/mule/test/extension/file/common/api/FileTestHarness.java [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+   */
+  boolean dirExists(String path) throws Exception;
+
+  /**
+   * @param path the path to test
+   * @return whether the given {@code path} exists and it's a file
+   */
+  boolean fileExists(String path) throws Exception;
+
+  boolean changeWorkingDirectory(String path) throws Exception;
```
Comment (pablokraan) [inline src/test/java/org/mule/test/extension/file/common/api/FileTestHarness.java]: Add javadoc

File: src/test/java/org/mule/test/extension/file/common/api/FileTestHarness.java [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+import org.junit.rules.ExpectedException;
+import org.junit.rules.TestRule;
+
+/**
+ * A {@link TestRule} which provides tools and environment for reusing the same test regardless of the remote server type (FTP,
+ * SFTP, etc) or the authentication method
+ *
+ * @since 1.0
+ */
+public interface FileTestHarness extends TestRule {
```
Comment (pablokraan) [inline src/test/java/org/mule/test/extension/file/common/api/FileTestHarness.java]: This class is not used on file connector, only on FTP and SFTP connectors.  Rename it

Comment (marianogonzalez) [reply src/test/java/org/mule/test/extension/file/common/api/FileTestHarness.java]: No. This is supposed to be a generic abstraction for testing a connector which operates over a file system. Tomorrow we may do samba, google drive or dropbox and still want to have an implementation of this class.

Comment (pablokraan) [reply src/test/java/org/mule/test/extension/file/common/api/FileTestHarness.java]: What is the expected result of calling getServerPort on a standard file system?

File: src/test/java/org/mule/test/extension/file/common/api/FileTestHarness.java [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+  /**
+   * @param path the path to test
+   * @return whether the given {@code path} exists and it's a file
+   * @throws Exception if the method fails
+   */
+  boolean fileExists(String path) throws Exception;
+
+  /**
+   * Moves the current working directory to the one signaled by {@code path}
+   * @param path the path to the new working directory
```
Comment (pablokraan) [inline src/test/java/org/mule/test/extension/file/common/api/FileTestHarness.java]: Add extra line between method and params descriptions

File: src/test/java/org/mule/test/extension/file/common/api/FileTestHarness.java [hunk 7]
```diff
@@ -0,0 +1,136 @@
...
+  /**
+   * @return the {@link ExpectedException} rule used for the current test
+   */
+  ExpectedError expectedError();
+
+  /**
+   * Creates a directory
+   *
+   * @param directoryPath the path to the directory to be created
+   *                      @throws Exception if the method fails
```
Comment (pablokraan) [inline src/test/java/org/mule/test/extension/file/common/api/FileTestHarness.java]: Use proper identation

File: src/test/java/org/mule/test/extension/file/common/api/FileTestHarness.java [hunk 8]
```diff
@@ -0,0 +1,136 @@
...
+  String BINARY_FILE_NAME = "binary.bin";
+  String HELLO_PATH = Paths.get("files", HELLO_FILE_NAME).toString();
+  String DEFAULT_FTP_HOST = "localhost";
+  String FTP_SERVER_BASE_DIR = Paths.get("target", "ftpserver").toString();
+  String WORKING_DIR_SYSTEM_PROPERTY = "workingDir";
+  String WORKING_DIR = "base";
+
+  /**
+   * Creates a test hello world file
+   * @throws Exception if the method fails
```
Comment (pablokraan) [inline src/test/java/org/mule/test/extension/file/common/api/FileTestHarness.java]: Add extra line

Comment (pablokraan) [inline src/test/java/org/mule/test/extension/file/common/api/FileTestHarness.java]: @throws Exception if the method fails > thanks for adding a so useful description

Comment (marianogonzalez) [reply src/test/java/org/mule/test/extension/file/common/api/FileTestHarness.java]: the description is as useful as your requirement ;)

File: src/test/java/org/mule/test/extension/file/common/api/FileTestHarness.java (hunk 5 above)
Comment (marianogonzalez) [reply src/test/java/org/mule/test/extension/file/common/api/FileTestHarness.java]: for the particular case of a local file system, none. For any other case, it does make sense

**Pull request merged**